pkg/cache/v2/cache.go

```diff
@@ -51,21 +52,66 @@ type Cache interface {
 	Fetch(context.Context, Request) (*Response, error)
 }
 
+type ResponseType interface {
```

Comment:
How about:
```suggestion
type Serializable interface {
```

Reply:
how about `Marshallable` because that's a already recognizable term in golang?

Reply:
Since we're changing the contract to be `GetDiscoveryResponse`, maybe `ResponseType` is ok or `ResponseIface`